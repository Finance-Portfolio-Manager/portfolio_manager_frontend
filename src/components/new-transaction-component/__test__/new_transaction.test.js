import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import NewTransaction from './../new_transaction';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = mount(<NewTransaction></NewTransaction>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<NewTransaction></NewTransaction>).toJSON();
    expect(tree).toMatchSnapshot();
})

describe("testing each function in new_transaction", ()=>{
    test("handleChange function on first input box", ()=>{
        console.log(wrapper.debug());
        const stockSymbol = wrapper.find("Form.Control").first();
        let changeAction = () =>{
            stockSymbol.simulate('change', {target: {value: 'TSLA'}})
        }
        expect(changeAction()).not.toThrow();
    })

    test("createNewTransaction function", ()=>{

    })

    test("getNewTransaction function", ()=>{

    })

    test("buyConfirm function", ()=>{

    })
})