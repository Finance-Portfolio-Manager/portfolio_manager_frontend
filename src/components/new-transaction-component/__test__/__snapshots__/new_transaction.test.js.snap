// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="wholeform"
>
  <form
    className=""
  >
    <div
      class="container"
    >
      <button
        className="btn btn-info"
        disabled={false}
        name="buy"
        onClick={[Function]}
        type="button"
        value="buysell"
      >
        Buy
      </button>
      <button
        className="btn btn-info"
        disabled={false}
        name="sell"
        onClick={[Function]}
        type="button"
        value="buysell"
      >
        Sell
      </button>
    </div>
    <div
      class="container"
    >
      <div>
        <label
          className="form-label"
        >
          Stock Symbol:
        </label>
        <input
          className="form-control"
          name="stocksymbol"
          onChange={[Function]}
        />
        <label
          className="form-label"
        >
          Quantity:
        </label>
        <input
          className="form-control"
          name="shareAmount"
          onChange={[Function]}
          type="number"
        />
        <label
          className="form-label"
        >
          Price:
        </label>
        <input
          className="form-control"
          name="sharePrice"
          onChange={[Function]}
          type="number"
        />
        <label
          className="form-label"
        >
          Date:
        </label>
        <input
          className="form-control"
          name="date"
          onChange={[Function]}
          type="date"
        />
        <label
          className="form-label"
        >
          Note:
        </label>
        <input
          className="form-control"
          name="note"
          onChange={[Function]}
        />
      </div>
    </div>
    <div
      class="container"
    >
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Add New Transaction
      </button>
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Test Get Transaction
      </button>
    </div>
  </form>
</div>
`;
